server/statistics/hot_cache.go

```diff
@@ -66,32 +66,28 @@ func (w *HotCache) Update(item *HotPeerStat) {
 }
 
 // RegionStats returns hot items according to kind
-func (w *HotCache) RegionStats(kind FlowKind) map[uint64][]*HotPeerStat {
+func (w *HotCache) RegionStats(kind FlowKind, minHotDegree int) map[uint64][]*HotPeerStat {
 	switch kind {
 	case WriteFlow:
-		return w.writeFlow.RegionStats()
+		return w.writeFlow.RegionStats(minHotDegree)
 	case ReadFlow:
-		return w.readFlow.RegionStats()
+		return w.readFlow.RegionStats(minHotDegree)
 	}
 	return nil
 }
 
 // RandHotRegionFromStore random picks a hot region in specify store.
-func (w *HotCache) RandHotRegionFromStore(storeID uint64, kind FlowKind, hotDegree int) *HotPeerStat {
-	if stats, ok := w.RegionStats(kind)[storeID]; ok {
-		for _, i := range rand.Perm(len(stats)) {
-			if stats[i].HotDegree >= hotDegree {
-				return stats[i]
-			}
-		}
+func (w *HotCache) RandHotRegionFromStore(storeID uint64, kind FlowKind, minHotDegree int) *HotPeerStat {
+	if stats, ok := w.RegionStats(kind, minHotDegree)[storeID]; ok && len(stats) > 0 {
+		return stats[rand.Intn(len(stats))]
```

Comment:
Do we need a rand seed?

Reply:
I think no need because rand seed is updated in `CreateServer`. P.S. the original `rand.Perm` is seed releated too.